builtinlib.ts

```diff
@@ -0,0 +1,142 @@
+import { Type } from './ast';
+import { RunTimeError } from './error_reporting';
+import { BOOL, NONE, NUM } from './utils';
+
+type BuiltinFunc = {
+  name: string
+  body: Function
+  typeSign: [Type[], Type]
```

Comment:
Haha it took me far too long to figure out `typeSign` is type signature because I kept reading sign as a word. I think typically `sig` is used here instead of `sign`.

Reply:
Oops, haha, my bad. I am a freshman in the U.S. Please forgive me for this.